### PR TITLE
Updates rust_test to use main.rs as the root when use_libtest_harness is false

### LINKED ISSUE #1516: `rust_test` should look for main.rs instead of lib.rs when use_libtest_harness is false

### PR DESCRIPTION
**Pull request by csmulhern**
Fixes #1516.

### CODE REVIEW

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Seems like a test needs to be updated to accompany this change? Otherwise seems good to me 😄 

(also, sorry for the delay!)

**Comment by csmulhern** [comment]
> Seems like a test needs to be updated to accompany this change? Otherwise seems good to me 😄
> 
> (also, sorry for the delay!)

Fixed!

(and no worries)

File: test/unit/use_libtest_harness/use_libtest_harness_test.bzl [hunk 1]
```diff
@@ -36,7 +36,10 @@ def _use_libtest_harness_test():
 
     rust_test(
         name = "mytest_noharness",
-        srcs = ["mytest_noharness.rs"],
+        srcs = [
```
Comment (UebelAndre) [inline test/unit/use_libtest_harness/use_libtest_harness_test.bzl:39]: Is this PR a breaking change? Do test targets with a single source file not using the test harness now require a `main.rs` or will they be able to function just as they did before?

**Review by UebelAndre — APPROVED** [review]
Looks good to me but one clarifying question about the impact of this change.

**Comment by csmulhern** [comment]
Test targets with a single source file not using the test harness function as they did before.

Test targets with multiple source files not using the test harness will break if they had not specified `crate_root` manually (i.e. they relied on the implicit use of `lib.rs` as the crate root, and defined a custom main function in their crate).

**Comment by UebelAndre** [comment]
> Test targets with a single source file not using the test harness function as they did before.
> 
> Test targets with multiple source files not using the test harness will break if they had not specified `crate_root` manually (i.e. they relied on the implicit use of `lib.rs` as the crate root, and defined a custom main function in their crate).

Would you be willing to add a test case for that? Just a copy of `mytest_noharness` that continues to use a single source file? 😅

**Comment by csmulhern** [comment]
Done!

**Review by UebelAndre — APPROVED** [review]
Thank you so much!

**Pull request merged**